File: vademecum/code-src/podstawowe_io/Operacje_na_plikach.cpp

// przy kompilacji konieczne jest dodanie:
//  -lboost_filesystem -lboost_system

#include <iostream>
#include <boost/filesystem.hpp>

int main() {
	// sprawdzenie czy plik istnieje
	boost::filesystem::path pFile( "/tmp/abc.txt" );
	if (boost::filesystem::is_regular_file(pFile))
		std::cout << "istnieje\n";
	else
		std::cout << "nie istnieje\n";
	
	// listowanie katalogu i sprawdzanie typów
	boost::filesystem::path pDir( "/tmp/" );
	boost::filesystem::directory_iterator end_iter;
	boost::filesystem::directory_iterator iter(pDir);
	
	for(; iter != end_iter; ++iter) {
		if (boost::filesystem::is_regular_file(iter->path())) {
			std::cout
				<< iter->path().filename().generic_string()
				<< " jest plikiem\n";
			// wypisujemy tylko nazwę pliku
		} else if (boost::filesystem::is_directory(iter->path())) {
			std::cout
				<< iter->path().generic_string()
				<< " jest katalogiem\n";
			// wypisuje,y pełną ścieżkę
		} else {
			std::cout
				<< iter->path().generic_string()
				<< " jest czymś innym\n";
		}
	}
	
	boost::filesystem::rename("/tmp/a.txt", "/tmp/b.txt");
	boost::filesystem::remove("/tmp/b.txt");
}

File: vademecum/code-src/podstawowe_io/Operacje_na_plikach.py

import os

# sprawdzenie czy plik istnieje
if os.path.isfile("/tmp/abc.txt"):
	print("istnieje")
else:
	print("nie istnieje")

# listowanie katalogu i sprawdzanie typów
dl = os.listdir("/tmp/")
print (dl)

for f in dl:
	if os.path.isfile(f):
		print("\"" + f + "\" jest plikiem")
	elif os.path.isdir(f):
		print("\"" + f + "\" jest katalogiem")
	else:
		print("\"" + f + "\" jest czymś innym")

# można także rekurencyjnie wraz z podkatalogami
for currDir, dirs, files in os.walk('/tmp'):
	print("podkatalogi \"" + currDir + "\":")
	for d in dirs:
		print("  " + os.path.join(currDir, d))
	
	print("pliki w \"" + currDir + "\":")
	for f in files:
		print("  " + os.path.join(currDir, f))

# zmiana nazwy
os.rename("/tmp/a.txt" "/tmp/b.txt")

# usuwanie
os.remove("/tmp/b.txt")
